tests/python/gaia-ui-tests/gaiatest/tests/functional/messages/test_sms_report.py

```diff
@@ -0,0 +1,77 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import expected, By, Wait
+
+from gaiatest.apps.base import Base
+
+from gaiatest import GaiaTestCase
+import time
+from gaiatest.apps.messages.app import Messages
+
+
+class TestSmsReport(GaiaTestCase):
+
+    def test_sms_report(self):
+        """This test sends a text message to itself. It waits for a reply message.
+           Then it checks the info in the report.
+        """
+        # Now here, then these locators will be in Report object
+        _text_type = (By.CSS_SELECTOR, 'p[data-l10n-id="message-type-sms"]')
+        _sent_time = (By.CSS_SELECTOR, 'span.sent-timestamp')
+        _received_time = (By.CSS_SELECTOR, 'span.received-timestamp')
+        _sender = (By.CSS_SELECTOR, 'a[data-name]')
+        
+        # Configuring the test
+        _text_message_content = "Automated Test %s" % str(time.time())
+        # _my_muber this way only during the local tests
+        _my_number = "+34644967894"
+        _text_type_expected = "Text Message (SMS)"
+        
+        # Sending the message
+        self.messages = Messages(self.marionette)
+        self.messages.launch()
+
+        new_message = self.messages.create_new_message(recipients=_my_number,
+                                                       message=_text_message_content)
+        self.message_thread = new_message.tap_send()
+        self.message_thread.wait_for_received_messages()
+
+        last_received_message = self.message_thread.received_messages[-1]
+        last_message = self.message_thread.all_messages[-1]
+
+        self.assertEqual(_text_message_content, last_received_message.text)
+        self.assertEqual(last_received_message.id, last_message.id)
+        
+        # Long Tap on received message and check the report: SMS sent received and time
+        activities = last_received_message.long_press_message()
+        time.sleep(5) # Will try to change this
+        report = activities.tap_report()
+
+        # Checking that the data in the received report is correct
+        # Type of message
+        #self.assertEqual(self._text_type.element.text(), _text_type_expected)
+        
+        # Sent and received dates are coherent
+        #self.assertEqual(self._received_time.text(), 'hora del mensaje enviado')
+        #self.assertEqual(self._received_time.text(), 'hora del mensaje recivido')
+
+        # From, sender is the right one
+        self.assertEqual(_my_number, _sender.text)
```

Comment:
Like discussed on IRC, this line is making the test fail. This is a python error. One way to solve is:
``` py
sender_value = self.marionette.find_element(*_sender).text
self.assertEqual(_my_number, sender_value)
```